src/vueWrapper.ts

```diff
@@ -1,35 +1,37 @@
-import { ComponentPublicInstance, nextTick, App } from 'vue'
+import {
+  nextTick,
+  App,
+  ComponentCustomProperties,
+  ComponentPublicInstance
+} from 'vue'
 import { ShapeFlags } from '@vue/shared'
 // @ts-ignore todo - No DefinitelyTyped package exists for this
 import pretty from 'pretty'
 
 import { config } from './config'
 import domEvents from './constants/dom-events'
 import { DOMWrapper } from './domWrapper'
-import {
-  FindAllComponentsSelector,
-  FindComponentSelector,
-  VueElement
-} from './types'
-import { createWrapperError } from './errorWrapper'
-import { find, matches } from './utils/find'
+import { VueElement } from './types'
 import { mergeDeep } from './utils'
 import { emitted, recordEvent } from './emit'
 import BaseWrapper from './baseWrapper'
-import WrapperLike from './interfaces/wrapperLike'
 
-export class VueWrapper<T extends ComponentPublicInstance>
-  extends BaseWrapper<T['$el']>
-  implements WrapperLike
-{
+export class VueWrapper<
+  T extends Omit<
+    ComponentPublicInstance,
+    '$emit' | keyof ComponentCustomProperties
+  > & {
+    $emit: (event: any, ...args: any[]) => void
+  } & ComponentCustomProperties = ComponentPublicInstance
+> extends BaseWrapper<Node> {
```

Comment:
TypeScript as usually sucks in co- and contravariance, that's why we need to relax check on `$emit` a bit (in other words `$emit` generated by `InstanceType<DefineComponent<...>>` which is `ComponentPublicInstance<...>` is not a subclass of `ComponentPublicInstance` (with default params) due to params type mismatch in `EmitFn` (the code of this helper type in vue-next is a bit insane)

But I wish someone told me why i need to decompose & put back `ComponentCustomProperties` - or `mount` will fail typecheck otherwise :confused: